pandas/core/arrays/base.py

```diff
@@ -856,7 +873,7 @@ def searchsorted(self, value, side="left", sorter=None):
         # 1. Values outside the range of the `data_for_sorting` fixture
         # 2. Values between the values in the `data_for_sorting` fixture
         # 3. Missing values.
-        arr = self.astype(object)
+        arr = cast(np.ndarray, self.astype(object))
```

Comment:
can you add a `TODO: overload astype so that cast is unnecessary` as a comment. In general, try to comment all casts to help future readers.